src/core/lender/tranche/restricted_operator.sol

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2020 Centrifuge
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.4.24;
+
+import "./operator.sol";
+
+// RestrictedOperator restricts the allowance of users
+// the allowanceActive flag actives the restriction
+// openAccess by default deactivated
+contract RestrictedOperator is Operator {
+
+    bool public supplyActive;
```

Comment:
We could make a own contract for the supplyActive and redeemActive flags.

Just an idea. We could use inheritance but better would be composition.
`ActiveOperator -> RestrictedOperator -> Operator`

it would allow to plug in a pipeline like:
`ActiveOperator  -> Operator`

But maybe over-engineering :)